src/urllib3/connection.py

```diff
@@ -370,8 +370,10 @@ def connect(self):
                 # We still support OpenSSL 1.0.2, which prevents us from verifying
                 # hostnames easily: https://github.com/pyca/pyopenssl/pull/933
                 or ssl_.IS_PYOPENSSL
-                # Python 3.6 can't disable commonName checks
-                or not hasattr(self.ssl_context, "hostname_checks_common_name")
```

Comment:
The original code is wrong. The feature depends on Python version and OpenSSL version.

```
if getattr(ssl, "HAS_NEVER_CHECK_COMMON_NAME", False):
    ctx.hostname_checks_common_name = False
```

This changes the internal attribute ``ctx._host_flags`` from ``4`` to ``36`` (``HOSTFLAG_NO_PARTIAL_WILDCARDS|HOSTFLAG_NEVER_CHECK_SUBJECT``). Internally it sets the OpenSSL flag ``X509_CHECK_FLAG_NEVER_CHECK_SUBJECT`` with ``X509_VERIFY_PARAM_set_hostflags()``.

Reply:
Thanks for pointing this out! I saw it earlier today and initially though that it was the only problem, but it's not. I will certainly use HAS_NEVER_CHECK_COMMON_NAME when reverting this.

(I also realize that setting `hostname_checks_common_name` only sets an OpenSSL flag, which is why I believe I'm doing something wrong.)